src/ast/main.cc

```diff
@@ -30,5 +30,8 @@ int main(int argc, char** argv)
   if (opt.ast)
     std::cout << peg::ast_to_s(ast) << std::endl;
 
+  if (opt.dump_path != "")
+    files::write(opt.dump_path + "/ast.txt", peg::ast_to_s(ast));
+
```

Comment:
This seems surprising, why isn't the output filename just the output filename, without appending something to it? Also, see other comment above on error handling and directories in paths.

Reply:
I intend to modify the test scripts to handle output redirect and remove the need for this option. But the goal of this PR is to integrate these tests with minimal changes to other tools.

Reply:
For now, the behavior of `--dump-path` is required by `update-dump.py` which we use to generate the test files we check against